File: clients/PHP-client/src/EnvClient.php
<?php

namespace SoliDB;

class EnvClient
{
    private Client $client;

    public function __construct(Client $client)
    {
        $this->client = $client;
    }

    public function list(): array
    {
        $res = $this->client->sendCommand('list_env_vars', ['database' => $this->client->getDatabase()]);
        return $res['data'] ?? [];
    }

    public function get(string $key): mixed
    {
        $res = $this->client->sendCommand('get_env_var', [
            'database' => $this->client->getDatabase(),
            'key' => $key
        ]);
        return $res['data'] ?? null;
    }

    public function set(string $key, mixed $value): void
    {
        $this->client->sendCommand('set_env_var', [
            'database' => $this->client->getDatabase(),
            'key' => $key,
            'value' => $value
        ]);
    }

    public function delete(string $key): void
    {
        $this->client->sendCommand('delete_env_var', [
            'database' => $this->client->getDatabase(),
            'key' => $key
        ]);
    }

    public function setBulk(array $vars): void
    {
        $this->client->sendCommand('set_env_vars_bulk', [
            'database' => $this->client->getDatabase(),
            'vars' => $vars
        ]);
    }
}

File: clients/Ruby-client/lib/solidb/env_client.rb
module SoliDB
  class EnvClient
    def initialize(client)
      @client = client
    end

    def list
      @client.send_command("list_env_vars", database: @client.database) || {}
    end

    def get(key)
      @client.send_command("get_env_var", database: @client.database, key: key)
    end

    def set(key, value)
      @client.send_command("set_env_var", database: @client.database, key: key, value: value)
      nil
    end

    def delete(key)
      @client.send_command("delete_env_var", database: @client.database, key: key)
      nil
    end

    def set_bulk(vars)
      @client.send_command("set_env_vars_bulk", database: @client.database, vars: vars)
      nil
    end
  end
end
